samples/simple_private_memory/simple_private_memory.cpp

```diff
@@ -33,14 +33,12 @@ using namespace cl::sycl;
  * hierarchical parallel_for_work_item context */
 static inline cl::sycl::id<1> get_global_id(cl::sycl::group<1>& group,
                                             cl::sycl::item<1>& item) {
-  cl::sycl::range<1> globalR = group.get_global_range();
+  cl::sycl::id<1> groupID = group.get();
   cl::sycl::range<1> localR = item.get_range();
-  cl::sycl::id<1> lID = item.get();
-  return cl::sycl::id<1>(group.get(0) * localR.get(0) + lID.get(0));
+  cl::sycl::id<1> localID = item.get();
+  return cl::sycl::id<1>(groupID.get(0) * localR.get(0) + localID.get(0));
```

Comment:
All these types are actually useles if you use `auto`... :-)

Auto-nice-ifying is the way to go to avoid these kinds of errors...